doc/notes/188_enhance_logging_ant.adoc

```diff
@@ -0,0 +1,245 @@
+message.
+
+The recommended location of this logging is in Sequence Verification.
+
+==== 4.3.2 Job-Level Statistics for Failures and Alarms
+
+For failed and alarmed jobs, the following statistics are required:
+
+* job type (name)
+* job identifier
+* event count
+* failure/alarm mode
+* event trace
+  ** event type (name)
+  ** event identifier
+  ** ordering
+
+=== 4.4 Event Logging
+
+Logging at the event level is largely infeasible and unnecessary in
+production configurations.  However, in anticipation of performance
+benchmarking, some event-level statistics shall be gathered in summary
+format.
+
+==== 4.4.1 Event Statistics
+
+=== 4.5 Protocol Verifier Statistics
+
+Several parameters are useful in the maintenance and support of the
+Protocol Verifier.
+
+The following statistics are reported as they happen.
+
+* worker registered
+* worker deregistered
+* unassigned job count exceeded
+  ** job count exceeded is reported once upon occurrence and then only
+     reported again if the job count goes below the threshold.
+
+The following statistic are reported periodically.
+
+* event count since start
+* event throughput
+* worker count
+* concurrent jobs at worker
+* concurrent jobs overall
+* unassigned jobs
+* unassigned job count exceeded
+
+This information shall be reported on a regular basis not exceeding once
+per second (1 Hertz).
+
+The recommended location of this logging is in Job Management.
+
+=== 4.6 Logger Hierarchy
+
+* The top-level logger shall be 'pv' (Protocol Verifier).
+* Each domain shall have a logger.
+  ** 'pv.jm':  Job Management
+  ** 'pv.aeo':  Audit Event Ordering
+  ** 'pv.sv':  Sequence Verification
+  ** 'pv.istore':  Invariant Store
+  ** 'pv.vg':  Verification Gateway
+* Each domain shall define at least 2 loggers:  'job' and 'event'.
+  ** 'pv.jm.job':  Job Management job logger
+  ** 'pv.jm.event':  Job Management event logger
+* Each domain may define lower-level loggers for specific purposes.
+  ** 'pv.sv.happy':  Sequence Verification Happy module (class)
+  ** 'pv.sv.unhappy':  Sequence Verification Unhappy module (class)
+* A logger shall be defined for benchmarking.
+  ** 'pv.benchmark':  throughput benchmark
+
+=== 4.7 Log Format
+
+There are two competing formats for log messages:  `LogFmt` and `JSON`.
+
+* LogFmt is simpler and cleaner looking in the source code.
+* JSON supports lists/arrays better.
+* LogFmt can be converted to JSON.
+
+In our source code, we will follow the LogFmt convention.  In our log
+formatter, we will convert to JSON.  Which can then be fed into Open
+Telemetry and just about any other log processing system.
+
+Read about LogFmt <<dr-6, here>>.
+
+== 5 Work Required
+
+. Remove the reporting terminator and use loggers for reporting.
+. Remove calls to AsyncLogger and use Logger with the new logger hierarchy.
+. Log a benchmark for each event as seen in Job Management and again in
+  Sequence Verification (tail end).
```

Comment:
What information would this **benchmark** contain

Reply:
This is purely to enable the measurement of the end-to-end time within the PV for each event.  This will be turned on by the test team (and me) when doing performance benchmarking.  It will be turned off in production.